### PR TITLE
Tno 2958 alert logs

### PR DESCRIPTION
**Pull request by fbarreta**
image

Notification logs implementation

### CODE REVIEW

File: api/net/Areas/Admin/Controllers/NotificationController.cs [hunk 1]
```diff
@@ -210,5 +211,22 @@ public async Task<IActionResult> SendToAsync(int id, long? contentId, string to)
...
     }
+
+    /// <summary>
+    /// Find all notifications for the dashboard that match the filter.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("dashboard")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<NotificationInstanceModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Report" })]
```
Comment (Fosol) [inline api/net/Areas/Admin/Controllers/NotificationController.cs:222]: `Notification` not `Report`

File: app/editor/src/features/admin/notifications/dashboard/NotificationsDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [filter]);
+
+  React.useEffect(() => {
+    findNotifications().then((data) => {
+      setNotificationOptions(
+        getNotificationOptions(data, filter.notificationId ? filter.notificationId : 0),
+      );
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Fosol) [inline app/editor/src/features/admin/notifications/dashboard/NotificationsDashboard.tsx:72]: Always include comment why you're not including all dependencies.

File: app/editor/src/features/admin/notifications/dashboard/utils/getLastSent.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { INotificationInstanceModel } from 'tno-core';
+
+import { formatDate } from './formatDate';
+
+export const getLastSent = (instance: INotificationInstanceModel) => {
```
Comment (Fosol) [inline app/editor/src/features/admin/notifications/dashboard/utils/getLastSent.ts:5]: Seeing duplicating with subscriber and editor code.  Ideally we'd move this to `tno-core` and update each app to use the shared identical method.

File: app/editor/src/features/admin/notifications/dashboard/utils/formatDate.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import moment from 'moment';
+/**
+ * @param {string | Date} date - The date to format
+ * @param {boolean} includeTime - Whether to include the time in the formatted date
+ * @returns {string} - The formatted date
+ * @description - This function takes a date and returns a formatted date string
+ */
+export const formatDate = (date: string | Date | undefined, includeTime?: boolean): string => {
```
Comment (Fosol) [inline app/editor/src/features/admin/notifications/dashboard/utils/formatDate.ts:8]: Seeing duplicating with subscriber and editor code.  Ideally we'd move this to `tno-core` and update each app to use the shared identical method.

File: app/editor/src/features/admin/notifications/dashboard/utils/getStatus.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { NotificationStatusName } from 'tno-core';
+
+/**
+ * Provides a message that represents the status.
+ * @param status The report status.
+ * @returns A message that represents the status.
+ */
+export const getStatus = (status?: NotificationStatusName) => {
```
Comment (Fosol) [inline app/editor/src/features/admin/notifications/dashboard/utils/getStatus.ts:8]: Seeing duplicating with subscriber and editor code.  Ideally we'd move this to `tno-core` and update each app to use the shared identical method.

File: app/editor/src/features/navbar/Menu.tsx [hunk 6]
```diff
@@ -97,6 +97,14 @@ export const Menu: React.FC = () => {
                   </MenuDropdownItem>
                 </Container>
               </NavDropdown>
+              <NavDropdown title="Notifications Building" id="basic-nav-dropdown">
```
Comment (Fosol) [inline app/editor/src/features/navbar/Menu.tsx:100]: Both these menu options should only be visible to administrator only.

**Review by Fosol — CHANGES_REQUESTED** [review]
Good first step to adding this dashboard.  There are some usability issues we'll need to address in another story.  Namely it'll be difficult to determine if a subscriber received an alert for a specific content item.

File: app/editor/src/features/admin/notifications/dashboard/utils/getLastSent.ts (hunk 3 above)
Comment (fbarreta) [reply app/editor/src/features/admin/notifications/dashboard/utils/getLastSent.ts:5]: This function is different than the report one. This one receive a INotificationInstanceModel, the report receives a IReportModel.

File: app/editor/src/features/admin/notifications/dashboard/utils/getStatus.ts (hunk 5 above)
Comment (fbarreta) [reply app/editor/src/features/admin/notifications/dashboard/utils/getStatus.ts:8]: This is also different, it returns a NotificationStatusName, and the report one returns a ReportStatusName.

File: libs/net/template/ReportExtensions.cs [hunk 7]
```diff
@@ -646,6 +646,7 @@ public static string GetSentimentColorScript(string[]? colors)
         // Use the index of the dataset to pick the colour.
         index = index >= 0 ? index : 0;
         var length = colors.Length;
+        if (length == 0) return "null";
```
Comment (Fosol) [inline libs/net/template/ReportExtensions.cs:649]: Adding this fix in to stop divide by zero issue when no colours are provided.

**Review by Fosol — APPROVED** [review]

**Pull request merged**